### PR TITLE
Fix issue with node envs (must explicitly set `duplex` option on Request constructror)

### PR DESCRIPTION
**Pull request by brettimus**
Alright. So. In environments that use node/undici, we need to conform to specifications APPARENTLY. 

- https://github.com/nodejs/node/issues/46221
- https://github.com/whatwg/fetch/pull/1457

***

The fetch specification is in a weird spot. 

Technically, when the body of a `RequestInit` is a ReadableStream, you **MUST** specify the property `duplex: "half"`

There will be future support for `duplex: "full"`, however this is not yet exposed. 

So really, the only option for `duplex` is `"half"` but we need to use it explicitly for future compatibility.

### CODE REVIEW

**Comment by brettimus** [comment]
tested by Nele on her/oscar's bun project, where the issue first came up

File: frontend/src/pages/RequestorPage/ResponsePanel.tsx [hunk 1]
```diff
@@ -426,7 +426,7 @@ function ResponseBody({
     }
 
     if (body?.type === "json") {
-      const prettyBody = JSON.stringify(JSON.parse(body.value), null, 2);
+      const prettyBody = JSON.stringify(safeParseJson(body.value), null, 2);
```
Comment (brettimus) [inline frontend/src/pages/RequestorPage/ResponsePanel.tsx:429]: shoehorned this in here, will be useful if we ever get a partial response back from a server for some reason ((happened to me, not sure why))

Comment (brettimus) [reply frontend/src/pages/RequestorPage/ResponsePanel.tsx:429]: the alternative here is that we crash the entire app

**Review by oscarvz — APPROVED** [review]
Can confirm it works on the bun project 👍

File: frontend/src/pages/RequestorPage/ResponsePanel.tsx [hunk 2]
```diff
@@ -660,3 +660,13 @@ function FailedWebsocket() {
     </div>
   );
 }
+
+const safeParseJson = (jsonString: string) => {
```
Comment (flenter) [inline frontend/src/pages/RequestorPage/ResponsePanel.tsx:664]: There's already an ~eerily similar~ exact same `safeParseJson` function in `src/utils/index.ts`. Why not import & use that?

**Review by flenter — COMMENTED** [review]
Oh it's been merged  😅  I didn't finish my review fast enough . I noticed one small thing though

**Pull request merged**